src/realm/alloc.hpp

```diff
@@ -64,14 +63,16 @@ class MemRef {
     void set_addr(char* addr);
 
 private:
-    char* m_addr;
-    ref_type m_ref;
+    char* m_addr = nullptr;
+    ref_type m_ref = 0;
 #if REALM_ENABLE_MEMDEBUG
     // Allocator that created m_ref. Used to verify that the ref is valid whenever you call
     // get_ref()/get_addr and that it e.g. has not been free'ed
     const Allocator* m_alloc = nullptr;
 #endif
 };
+static_assert(std::is_trivially_destructible_v<MemRef>);
+static_assert(std::is_trivially_copyable_v<MemRef>);
```

Comment:
```suggestion
static_assert(std::is_trivially_copyable_v<MemRef>);
```

trivially copyable implies/requires trivially destructible, so you don't need to check both. (Technically `std::is_trivially_copyable` also checks that the destructor is public, which isn't needed for TriviallyCopyable, but that isn't something we care about asserting here)